timer/timer.go

```diff
@@ -0,0 +1,120 @@
+package timer
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/nivista/steady/.gen/protos/common"
+	"github.com/nivista/steady/internal/.gen/protos/messaging"
+)
+
+type (
+	// Timer represents the configuration for the execution of a recurring task.
+	Timer struct {
+		executer  executer
+		scheduler scheduler
+	}
+
+	// Progress records the progress of a recurring task.
+	Progress struct {
```

Comment:
Why does progress need to be exported?

Reply:
In this iteration the timer is stateless. Progress is held outside the timer and is passed to the run function. The progress could also not be stateless but idk, it's like what you said about how timers could be smart or stupid. My timers are stupid and stateless because that made the most sense to me.

Reply:
I should also note that Progress is basically an exact replica of a proto type. Could i just use the proto type?